datafusion/optimizer/src/decorrelate_scalar_subquery.rs

```diff
@@ -101,16 +101,17 @@ impl OptimizerRule for DecorrelateScalarSubquery {
 
                 let (subqueries, other_exprs) =
                     self.extract_subquery_exprs(predicate, optimizer_config)?;
-                let optimized_plan = LogicalPlan::Filter(Filter {
-                    predicate: predicate.clone(),
-                    input: Arc::new(optimized_input),
-                });
+
                 if subqueries.is_empty() {
                     // regular filter, no subquery exists clause here
+                    let optimized_plan = LogicalPlan::Filter(Filter {
+                        predicate: predicate.clone(),
+                        input: Arc::new(optimized_input),
+                    });
                     return Ok(optimized_plan);
                 }
 
-                // iterate through all exists clauses in predicate, turning each into a join
+                // iterate through all subqueries in predicate, turning each into a join
```

Comment:
Thanks for fixing my messy docs and comments everywhere!

Reply:
it's very well commented! the comments were extremely helpful